packages/wrapped-keys/src/lib/litActions.ts

```diff
@@ -49,43 +43,95 @@ const DEFAULT_GAS_PRICE = '50'; // in gwei
 
     const tx = {
         to: unsignedTransaction.toAddress,
-        value: ethers.utils.parseEther(unsignedTransaction.value),
+        value: ethers.utils.hexlify(ethers.utils.parseEther(unsignedTransaction.value)),
         chainId: unsignedTransaction.chainId,
-        gasPrice: ethers.utils.parseUnits(gasPrice, 'gwei'),
-        gasLimit,
         data: unsignedTransaction.dataHex,
         nonce,
     };
 
+    if (unsignedTransaction.gasPrice) {
+        tx.gasPrice = ethers.utils.parseUnits(unsignedTransaction.gasPrice, 'gwei');
+    } else {
+        tx.gasPrice = await Lit.Actions.runOnce(
+            { waitForResponse: true, name: 'gasPrice' },
+            async () => {
+                const rpcUrl = await Lit.Actions.getRpcUrl({ chain: unsignedTransaction.chain });
+                const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+    
+                try {
+                    const gasPrice = await provider.getGasPrice();
+                    return ethers.utils.hexlify(gasPrice);
+                } catch (err) {
+                    const errorMessage = 'Error: When getting gas price- ' + err.message;
+                    Lit.Actions.setResponse({ response: errorMessage });
+                    return;
+                }
+            }
+        )
+    }
+
+    if (unsignedTransaction.gasLimit) {
+        tx.gasLimit = ethers.utils.hexlify(unsignedTransaction.gasLimit)
+    } else {
+        tx.gasLimit = await Lit.Actions.runOnce(
+            { waitForResponse: true, name: 'gasLimit' },
+            async () => {
+                const rpcUrl = await Lit.Actions.getRpcUrl({ chain: unsignedTransaction.chain });
+                const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+                
+                try {
+                    const gasLimit = await provider.estimateGas(tx);
+                    return ethers.utils.hexlify(gasLimit);
+                } catch (err) {
+                    const errorMessage = 'Error: When estimating gas- ' + err.message;
+                    Lit.Actions.setResponse({ response: errorMessage });
+                    return;
+                }
+            }
+        )
+    }
+
     console.log('tx');
     console.log(tx);
 
-    try {
-        const signedTx = await wallet.signTransaction(tx);
+    const signedTx = await Lit.Actions.runOnce(
```

Comment:
Since only once node is signing the tx why not broadcast in the same `runOnce` block?

Reply:
I've removed this code when merging in your latest changes